pallets/automation-time/src/tests.rs

```diff
@@ -1017,30 +1124,101 @@ fn missed_tasks_removes_completed_tasks() {
 #[test]
 fn trigger_tasks_completes_some_native_transfer_tasks() {
 	new_test_ext(START_BLOCK_TIME).execute_with(|| {
-		get_funds(ALICE);
-		let current_funds = Balances::free_balance(ALICE);
+		get_funds(AccountId32::new([ALICE; 32]));
+		let current_funds = Balances::free_balance(AccountId32::new([ALICE; 32]));
 		let transfer_amount = 1;
 
 		add_task_to_task_queue(
 			ALICE,
 			vec![40],
 			vec![SCHEDULED_TIME],
-			Action::NativeTransfer { sender: ALICE, recipient: BOB, amount: transfer_amount },
+			Action::NativeTransfer { sender: AccountId32::new([ALICE; 32]), recipient: AccountId32::new([BOB; 32]), amount: transfer_amount },
 		);
 		add_task_to_task_queue(
 			ALICE,
 			vec![50],
 			vec![SCHEDULED_TIME],
-			Action::NativeTransfer { sender: ALICE, recipient: BOB, amount: transfer_amount },
+			Action::NativeTransfer { sender: AccountId32::new([ALICE; 32]), recipient: AccountId32::new([BOB; 32]), amount: transfer_amount },
+		);
+
+		LastTimeSlot::<Test>::put((LAST_BLOCK_TIME, LAST_BLOCK_TIME));
+		System::reset_events();
+
+		AutomationTime::trigger_tasks(120_000);
+
+		assert_eq!(Balances::free_balance(AccountId32::new([ALICE; 32])), current_funds - (transfer_amount * 2));
+		assert_eq!(Balances::free_balance(AccountId32::new([BOB; 32])), transfer_amount * 2);
+	})
+}
+
+#[test]
+fn trigger_tasks_completes_some_xcmp_tasks() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		let task_id1 = add_task_to_task_queue(
+			ALICE,
+			vec![40],
+			vec![SCHEDULED_TIME],
+			Action::XCMP {
+				para_id: PARA_ID.try_into().unwrap(),
+				call: vec![3, 4, 5],
+				weight_at_most: 100_000,
+			},
+		);
+
+		LastTimeSlot::<Test>::put((LAST_BLOCK_TIME, LAST_BLOCK_TIME));
+		System::reset_events();
+
+		AutomationTime::trigger_tasks(120_000);
+
+		assert_eq!(
+			sent_xcm(),
+			vec![(
+				(1, Junction::Parachain(PARA_ID.into())).into(),
+				Xcm(vec![Transact {
+					origin_type: OriginKind::SovereignAccount,
```

Comment:
wait, shouldn't this origin_type be type Native? I had trouble getting SovereignAccount to work at least while doing testing locally. This could also totally be a test only thing, but would be good to know why i think?

Reply:
yea, this should be a `SovereignAccount`.  I was playing around and broke it.  Fixed now.